src/backend/opencl/morph3d_impl.hpp

```diff
@@ -18,31 +18,32 @@ using af::dim4;
 
 namespace opencl
 {
-
 template<typename T, bool isDilation>
 Array<T> morph3d(const Array<T> &in, const Array<T> &mask)
 {
     const dim4 mdims    = mask.dims();
 
     if (mdims[0]!=mdims[1] || mdims[0]!=mdims[2])
-        AF_ERROR("Only cube masks are supported in opencl morph currently", AF_ERR_SIZE);
+        OPENCL_NOT_SUPPORTED("Only cube masks are supported in opencl morph currently");
```

Comment:
It looks like we don't support this functionality in CUDA or OpenCL. No need to specify that a certain backend doesn't support this. Also it would be better if you reworded it to:

"Cube masks are not supported."